Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -1183,14 +1183,14 @@ private List<String> getCanonicalFileNamePartsAndSanitizeAbsoluteURIs(Workpiece
                     physicalDivision.getMediaFiles().put(entry.getKey(), mediaFile);
                 }
                 String fileCanonical = subfolder.getCanonical(mediaFile);
-                if ("".equals(unitCanonical)) {
+                if (unitCanonical.isEmpty()) {
```

Comment:
Calling `equals` on a non-empty String literal instead of String variable (that could be `null`) implicitly prevents `NullPointerExceptions`. 

Since this null-save comparison was made here and because I cannot say whether it was done to explicitly deal with null values, I would prefer to guarantee the same behaviour and either add an explicit null-check
`Objects.isNull(unitCanonical) || unitCanonical.isEmpty()`
or use the null-save 
`StringUtils.isBlank(unitCanonical)`
here.

(Same goes for the next to changes, see below)

Reply:
Good catch. I will change it to `StringUtils.isBlank()` call on all 3 mentioned places.